src/pages/index.js

```diff
@@ -22,96 +34,233 @@ const profileDescriptionInput = profileEditModal.querySelector(
   "#profile-description-input"
 );
 
-//Add New Card
-const addCardModal = document.querySelector("#add-card-modal");
+// Edit Avatar
+const avatarEditBtn = document.querySelector("#avatar-edit-button");
+const avatarForm = document.forms["modal__form_avatar"];
+
+const profileImage = document.querySelector(".profile__image");
+
+// Add New Card
 const addCardForm = document.forms["card-form"];
 const addCardBtn = document.querySelector(".profile__add-button");
 
-const renderCard = (cardData) => {
-  const card = createCard(cardData);
-  section.addItem(card);
-};
+// UserInfo
+const user = new UserInfo(
+  ".profile__title",
+  ".profile__description",
+  ".profile__image"
+);
+
+api
+  .getUserData()
+  .then((profileData) => {
+    if (profileData) {
+      user.setUserInfo(profileData.name, profileData.about);
+      user.changeAvatar(profileData.avatar);
+    }
+  })
+  .catch((err) => console.log("Error loading user info:", err));
 
 /* -------------------------------------------------------------------------- */
 /*                                  Popups                                    */
 /* -------------------------------------------------------------------------- */
-//Edit Profile Popup
+
+// Edit Profile Popup
 const editProfilePopup = new PopupWithForm(
   "#profile-edit-modal",
-  handleProfileEditSubmit
+  (profileData) => {
+    const submitButton = profileEditForm.querySelector(".modal__save");
+    const initialButtonText = submitButton.textContent;
+    submitButton.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```